File: GanttChartLightLibraryDemos/Demos/Samples/WPF-CSharp/GanttChartDataGrid/Recurrence/RecurrentGanttChartItem.cs
using System;
using DlhSoft.Windows.Controls;

namespace Demos.WPF.CSharp.GanttChartDataGrid.Recurrence
{
    public class RecurrentGanttChartItem : GanttChartItem
    {
        private RecurrenceType recurrenceType = RecurrenceType.Weekly;
        public RecurrenceType RecurrenceType { get { return recurrenceType; } set { recurrenceType = value; OnPropertyChanged("RecurrenceType"); } }

        private int occurrenceCount = 1;
        public int OccurrenceCount { get { return occurrenceCount; } set { occurrenceCount = Math.Max(1, value); OnPropertyChanged("OccurrenceCount"); } }
    }
}

File: GanttChartLightLibraryDemos/Demos/Samples/WPF-VisualBasic/GanttChartDataGrid/Recurrence/RecurrentGanttChartItem.vb
Imports DlhSoft.Windows.Controls

Public Class RecurrentGanttChartItem
    Inherits GanttChartItem

    Private recurrenceTypeValue As RecurrenceType = RecurrenceType.Weekly
    Public Property RecurrenceType() As RecurrenceType
        Get
            Return recurrenceTypeValue
        End Get
        Set(value As RecurrenceType)
            recurrenceTypeValue = value
            OnPropertyChanged("RecurrenceType")
        End Set
    End Property

    Private occurrenceCountValue As Integer = 1
    Public Property OccurrenceCount() As Integer
        Get
            Return occurrenceCountValue
        End Get
        Set(value As Integer)
            occurrenceCountValue = Math.Max(1, value)
            OnPropertyChanged("OccurrenceCount")
        End Set
    End Property
End Class
